rasterio/io.py

```diff
@@ -105,36 +106,36 @@ class WindowMethodsMixin(object):
     properties: `transform`, `height` and `width`
     """
 
-    def window(self, left, bottom, right, top, boundless=True, precision=6):
+    def window(self, left, bottom, right, top, precision=6, **kwargs):
         """Get the window corresponding to the bounding coordinates.
 
         Parameters
         ----------
-        left : float
+        left: float
             Left (west) bounding coordinate
-        bottom : float
+        bottom: float
             Bottom (south) bounding coordinate
-        right : float
+        right: float
             Right (east) bounding coordinate
-        top : float
+        top: float
             Top (north) bounding coordinate
-        boundless: boolean, optional
-            If boundless is False, window is limited
-            to extent of this dataset.
-        precision : int, optional
+        precision: int, optional
             Number of decimal points of precision when computing inverse
             transform.
+        kwargs: mapping
+            For backwards compatibility: absorbs deprecated keyword args.
 
         Returns
         -------
         window: Window
         """
-
+        if 'boundless' in kwargs:
+            warnings.warn("boundless keyword arg should not be used",
+                          RasterioDeprecationWarning)
```

Comment:
Not a blocker but should we write a test to cover this? The value of a test here is minimal but... coverage.